### DISCUSSION TITLE
Is GLOBALS and Compiler thread safe?

### DISCUSSION THREAD

**Discussion by eighty4** [swc-project/swc#9188] [General]
I want to use multiple Compiler instances in a parallel and I'm only familiar with the `GLOBALS.set(&Default::default() || { try_with_handler })` setup used in the SWC crate examples. Would spans of multiple builds affect other builds or would it be thread safe? Suggestions to read in the SWC sources to understand what GLOBALS and spans are/do would be helpful.

**Comment by kdy1** [comment]
Yeas, those are thread-safe

**Reply by kdy1** [reply]
If a type is `Send + Sync` in rust, it's thread safe.

**Comment by eighty4** [comment]
Awesome, thanks.
